### PR TITLE
Gem metadata backfill

### PR DESCRIPTION
**Pull request by jimmycuadra**
Continued from #858, this branch adds a new Rake task, `gemcutter:metadata:backfill` for backfillilng old gem versions with metadata. I based it on the Rake task written for the checksum backfill, so it'll be able to run in parallel against multiple shards.

The use of `Pusher` to extract the specification from a gem file is a bit odd, but I didn't find any code that did that anywhere else and thought it best not to go out of scope and factor that code out of `Pusher`, so I just used it here directly. It's migration code anyway, and can be removed when the migration is done.

### CODE REVIEW

**Comment by jimmycuadra** [comment]
Currently blocked by #914.

**Comment by sferik** [comment]
@jimmycuadra That was just deployed by @arthurnn, so this should be unblocked now.

**Comment by arthurnn** [comment]
(@sferik, I didnt deploy yet.. will do it this afternoon, and come back to this)

**Comment by sferik** [comment]
Is it deployed yet?

**Comment by arthurnn** [comment]
deployed!

File: lib/metadata_backfill.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+  def initialize(version)
+    @version = version
+  end
+
+  def backfill
+    file = dir.files.get(key) or return
+    pusher = Pusher.new(nil, StringIO.new(file.body))
+    pusher.pull_spec
+    version.update_attributes(metadata: pusher.spec.metadata || {})
```
Comment (arthurnn) [inline lib/metadata_backfill.rb]: s/update_attributes/update

Comment (arthurnn) [inline lib/metadata_backfill.rb]: Also, do we want to push a empty `{}` ? or should we just not update the attribute if `pusher.spec.metadata` is nil

**Comment by jimmycuadra** [comment]
Rebased and made both of those changes!

**Comment by arthurnn** [comment]
@jimmycuadra thanks for you work.
I will be merging it, this weekend so I can run the jobs and watch it closely.

**Comment by jimmycuadra** [comment]
Any updates on this?

**Comment by jimmycuadra** [comment]
Still hoping to have this job run...

**Comment by arthurnn** [comment]
Thanks @jimmycuadra 

Will be running this, and update this once is done.

**Comment by arthurnn** [comment]
50% done so far.. probably tomorrow we will be 100%. will update this.

**Comment by arthurnn** [comment]
DONE.

**Comment by jimmycuadra** [comment]
Thank you very much!! :grinning:

**Pull request merged**